packages/dcos-integration-test/extra/test_ucr.py

```diff
@@ -210,3 +253,48 @@ def test_if_ucr_pods_can_be_deployed_with_image_digest(dcos_api_session):
     with dcos_api_session.marathon.deploy_pod_and_cleanup(pod_definition):
         # Trivial app if it deploys, there is nothing else to check
         pass
+
+
+def test_if_ucr_pods_can_be_deployed_with_auto_cgroups(dcos_api_session):
+    """Marathon pods inside ucr deployment integration test.
+
+    This test launches a marathon ucr pod to verify the CPU and memory
+    cgroups subsystems can be automatically loaded by the agent and the
+    container specific cgroups can be correctly mounted inside the container.
+
+    Please note that the `cgroups/all` option has been specified in the agent
+    `--isolation` flag in `calculate_mesos_isolation()` which means the agent
+    will automatically load all the local enabled cgroups subsystems. And the
+    resources read from the container specific cgroups include both the two
+    container's resources (0.2 CPU and 64MB memory) and the default executor's
+    resources (0.1 CPU and 32MB memory).
+    """
+    test_uuid = uuid.uuid4().hex
+    pod_definition = {
+        'id': '/integration-test-pods-{}'.format(test_uuid),
+        'scaling': {'kind': 'fixed', 'instances': 1},
+        'environment': {'PING': 'PONG'},
+        'containers': [
+            {
+                'name': 'container1',
+                'resources': {'cpus': 0.1, 'mem': 32},
+                'image': {'kind': 'DOCKER', 'id': 'library/alpine'},
+                'exec': {
+                    'command': {
+                        'shell': 'test `cat /sys/fs/cgroup/memory/memory.soft_limit_in_bytes` = 100663296 && '
+                                 'test `cat /sys/fs/cgroup/cpu/cpu.shares` = 307'
```

Comment:
Same here - I don't know whether this test is portable to different systems. Isn't cpu.shares relative to whatever else is running in the cgroup?